src/Rpc/NodeGuardService.cs

```diff
@@ -693,18 +694,47 @@ public override async Task<AddLiquidityRuleResponse> AddLiquidityRule(AddLiquidi
         var rules = await _liquidityRuleRepository.GetByNodePubKey(node.PubKey);
         var rule = rules.FirstOrDefault(r => r.ChannelId == request.ChannelId);
 
-        var wallet = await _walletRepository.GetById(request.WalletId);
-        if (wallet == null)
+        var swapWallet = await _walletRepository.GetById(request.SwapWalletId);
+        if (swapWallet == null)
         {
-            throw new RpcException(new Status(StatusCode.NotFound, "Wallet not found"));
+            throw new RpcException(new Status(StatusCode.NotFound, "Swap wallet not found"));
+        }
+
+        if (request.IsReverseSwapWalletRule)
+        {
+            if (!request.HasReverseSwapWalletId)
```

Comment:
why is this necessary? maybe is better to compare `request.ReverseSwapWalletId == null` and have one less field that might not be updated

Reply:
This is more of a double check, if someone forgot to remove when changing and both fields were populated this field would be essential to determine the way to execute the swap.

Reply:
I meant that the field seems unnecessary, not the check. But is just to be nitpicky